Doc/library/sys.monitoring.rst

```diff
@@ -0,0 +1,268 @@
+:mod:`sys.monitoring` --- Execution event monitoring
+====================================================
+
+..  module:: sys.monitoring
+
+    :synopsis: Access and control event monitoring
+
+   .. note::
+
+      :mod:`sys.monitoring` is a namespace within the `sys` module,
+      not an independent module, so ``import sys.monitoring`` will not
+      work.
+
+-----------------
+
+This namespace provides access to the functions and constants necessary to
+activate and control event monitoring.
+
+As programs execute, events occur that might be of interest to tools that
+monitor execution. The :mod:`sys.monitoring` namespace provides means to
+receive callbacks when events of interest occur.
+
+The monitoring API consists of three components:
+
+* Tool identifiers
+* Events
+* Callbacks
+
+Tool identifiers
+----------------
+
+A tool identifier is an integer and associated name.
+Tool identifiers are used to discourage tools from interfering with each
+other and to allow multiple tools to operate at the same time.
+Currently tools are completely independent and cannot be used to
+monitor each other. This restriction may be lifted in the future.
+
+Before registering or activating events, a tool should choose an identifier.
+Identifiers are integers in the range 0 to 5.
+
+To register a tool::
+
+  sys.monitoring.use_tool_id(id: range(6), name:str) -> None
+  sys.monitoring.free_tool_id(id: range(6)) -> None
+  sys.monitoring.get_tool(id: range(6)) ->  str | None
+
+
+``sys.monitoring.use_tool_id`` raises a ``ValueError`` if ``id`` is in use.
+``sys.monitoring.get_tool`` returns the name of the tool if ``id`` is in use,
+otherwise it returns ``None``.
```

Comment:
```suggestion
:py:func:`sys.monitoring.use_tool_id` raises a ``ValueError`` if ``id`` is in use.
:py:func:`sys.monitoring.get_tool` returns the name of the tool if ``id`` is in use,
otherwise it returns ``None``.
```